File: back_chat/src/main/java/com/sovereingschool/back_chat/Models/ClaseChat.java
package com.sovereingschool.back_chat.Models;

import java.util.List;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class ClaseChat {
    @Id
    private Long idClase;

    private Long idCurso;

    private List<MensajeChat> mensajes;

}

File: front/src/app/models/ClaseChat.ts
import { MensajeChat } from './MensajeChat';

export class ClaseChat {
	public id_clase: number;

	public id_curso: number;

	public nombre_clase: string;

	public mensajes: MensajeChat[];

	constructor(_id_clase: number, _id_curso: number, _nombre_clase: string, _mensajes: MensajeChat[]) {
		this.id_clase = _id_clase;
		this.id_curso = _id_curso;
		this.nombre_clase = _nombre_clase;
		this.mensajes = _mensajes;
	}
}
